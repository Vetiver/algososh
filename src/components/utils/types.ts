import { ElementStates } from "../../types/element-states";

export type TStringArray = string[];

export type TSortingStringArray = {
  value: string | number;
  type: ElementStates;
};

export type TSortingNumberArray = {
  value: number;
  type: ElementStates;
};



export type TSortingArray = {
  value: any;
  type: ElementStates;
};
