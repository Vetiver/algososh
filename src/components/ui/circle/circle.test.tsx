import renderer from "react-test-renderer";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";



describe("circle", () => {
  it('empty circle', () => {
    const emptyCircle = renderer.create(<Circle />).toJSON();
    expect(emptyCircle).toMatchSnapshot()
  })
  it('circle with text', () => {
    const circleWithText = renderer.create(<Circle letter='ÐŸ' />).toJSON();
    expect(circleWithText).toMatchSnapshot()
  })
  it('circle with head', () => {
    const circleWithHead = renderer.create(<Circle head={<Circle />} />).toJSON();
    expect(circleWithHead).toMatchSnapshot()
  })
  it('circle with tail', () => {
    const circleWithTail = renderer.create(<Circle tail={<Circle />} />).toJSON();
    expect(circleWithTail).toMatchSnapshot()
  })
  it('circle with index', () => {
    const circleWithIndex = renderer.create(<Circle index={2} />).toJSON();
    expect(circleWithIndex).toMatchSnapshot()
  })
  it('circle is small', () => {
    const circleIsSmall = renderer.create(<Circle isSmall={true} />).toJSON();
    expect(circleIsSmall).toMatchSnapshot()
  })
  it('circle is default', () => {
    const circleIsDefault = renderer.create(<Circle state={ElementStates.Default} />).toJSON();
    expect(circleIsDefault).toMatchSnapshot()
  })
  it('circle is changing', () => {
    const circleIsChanging = renderer.create(<Circle state={ElementStates.Changing} />).toJSON();
    expect(circleIsChanging).toMatchSnapshot()
  })
  it('circle is modified', () => {
    const circleIsModified = renderer.create(<Circle state={ElementStates.Modified} />).toJSON();
    expect(circleIsModified).toMatchSnapshot()
  })
}) 