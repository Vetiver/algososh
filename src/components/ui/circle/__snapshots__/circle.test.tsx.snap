// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`circle circle is changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   changing"
    data-testid="stringCircle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="contentCircle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`circle circle is default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="stringCircle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="contentCircle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`circle circle is modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   modified"
    data-testid="stringCircle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="contentCircle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`circle circle is small 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle  small default"
    data-testid="circleTarget"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="contentCircle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`circle circle with head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="head"
      />
      <div
        className="circle   default"
        data-testid="stringCircle"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="contentCircle"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-testid="circleIndex"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="tail"
      />
    </div>
  </div>
  <div
    className="circle   default"
    data-testid="stringCircle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="contentCircle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`circle circle with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="stringCircle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="contentCircle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  >
    2
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-testid="tail"
  />
</div>
`;

exports[`circle circle with tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="stringCircle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="contentCircle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-testid="head"
      />
      <div
        className="circle   default"
        data-testid="stringCircle"
      >
        <p
          className="text text_type_circle text_color_input letter"
          data-testid="contentCircle"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-testid="circleIndex"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-testid="tail"
      />
    </div>
  </div>
</div>
`;

exports[`circle circle with text 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="stringCircle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="contentCircle"
    >
      ÐŸ
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;

exports[`circle empty circle 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="head"
  />
  <div
    className="circle   default"
    data-testid="stringCircle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="contentCircle"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circleIndex"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="tail"
  />
</div>
`;
